provider/pihole/pihole.go

```diff
@@ -74,25 +74,40 @@ func (p *PiholeProvider) Records(ctx context.Context) ([]*endpoint.Endpoint, err
 
 // ApplyChanges implements Provider, syncing desired state with the Pi-hole server Local DNS.
 func (p *PiholeProvider) ApplyChanges(ctx context.Context, changes *plan.Changes) error {
-	// Handle deletions first - there are no endpoints for updating in place.
+	// Handle pure deletes first.
 	for _, ep := range changes.Delete {
 		if err := p.api.deleteRecord(ctx, ep); err != nil {
 			return err
 		}
 	}
+
+	// Handle updated state - there are no endpoints for updating in place.
+	updateNew := make(map[string]*endpoint.Endpoint)
+	for _, ep := range changes.UpdateNew {
+		updateNew[ep.DNSName] = ep
```

Comment:
What if `changes.UpdateNew` has multiple endpoints with the same `DNSName`? Shouldn't you put the `RecordType` in the map key?

Reply:
Good catch! PiHole doesn't allow two records of the same RecordType with the same DNS name, but it does allow two records with the same DNS name if the record types are different. Fixed